runtime/gc_glue_java/ConcurrentMarkingDelegate.hpp

```diff
@@ -355,6 +355,14 @@ class MM_ConcurrentMarkingDelegate
 
 	bool setupClassScanning(MM_EnvironmentBase *env);
 
+	/**
+	 * Allocate an obj (type java/lang/object) at allocAddr.
+	 *
+	 * Specific use for this is to allocate an obj the end of the TLH. This is done to batch mark the cache for SATB, which requires
+	 * the collector to know the starting addr of the last obj in the cache.
+	 */
+	void *initializeMinimumSizeObject(MM_EnvironmentBase *env, uint8_t *allocAddr);
```

Comment:
This method (even an earlier version) does not allocate (it occured earlier before this call). It's more clear with the new name, but still I'd adjust the comment like

Initialize(or create?) an object of minimal object size at given addr. Specifically, java/lang/object is used.

arg allocAddr [in] address at which the object is created


You can keep the extra comment what  this may be needed for, although it may not be the only future used. Again I'd use initialize/create instead of allocate in that comment.

Also, we don't need return value.

Reply:
Also, this method does not necessarily need to be in ConcurrentMarkingDelegate class. A more basic Delegate (ObjectModelDelegate) or some other class (Extensions?) would be a better fit.

Reply:
Comment has been updated and return is removed

Reply:
>Also, this method does not necessarily need to be in ConcurrentMarkingDelegate class. A more basic language Delegate (ObjectModelDelegate?) or some other class (Extensions?) would be a better fit.

I agree, the method is more generic now

Reply:
If we move it to `GC_ObjectModelDelegate`,  do we indent to use this pattern:
 `_extensions->objectModel.initializeMinimumSizeObject(..)` calls -> `ObjectModelBase:: initializeMinimumSizeObject(..)` calls-> `GC_ObjectModelDelegate::initializeMinimumSizeObject(..)`

Reply:
I think that's the right way, but I believe it's not a problem (the base wrapper and it's usage is all in the next OMR change)